src/middlewared/middlewared/plugins/pool_/attach_disk.py

```diff
@@ -62,3 +64,25 @@ async def attach(self, job, oid, options):
         await job.wrap(extend_job)
 
         self.middleware.create_task(self.middleware.call('disk.swaps_configure'))
+
+        if vdev['type'] in ('RAIDZ1', 'RAIDZ2', 'RAIDZ3'):
+            while True:
```

Comment:
Should we have some sort of upper bound limit to timeout in case something goes wrong with the job itself ?

Reply:
@sonicaj no, this process can go for dozens of days in case of a very large pool